src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs

```diff
@@ -499,6 +499,53 @@ public static bool TryParse(string? value, out MyTryParsableRecord? result)
             }
         }
 
+        private class MyBindAsyncTypeThatThrows
+        {
+            public static ValueTask<object?> BindAsync(HttpContext context)
```

Comment:
What's the behavior like if the signature does not match? e.g. `ValueTask BindAsync(HttpContext context)`?

Reply:
It won't find it

Reply:
I could fail if it found the method but had the wrong return type?

Reply:
Toss it in the Analyzer pile?

Reply:
Can we just look for `ValueTask<{type}>` in `FindBindAsyncMethod`? What's the benefit of allowing any object to be returned? I get that we're going to box anyway for structs, but might as well lean on the type-safety checks of the compiler.

We could even throw at startup if a type is used with a `BindAsync(HttpContext)` method with the wrong return type.